framework/src/modules/dpos_v2/module.ts

```diff
@@ -596,18 +590,40 @@ export class DPoSModule extends BaseModule {
 				height + 1 - 2 * this._moduleConfig.roundLength,
 				this._moduleConfig.roundLength,
 			);
-			const standbyDelegates = selectStandbyDelegates(
+			standbyDelegates = selectStandbyDelegates(
 				snapshot.delegateWeightSnapshot,
 				randomSeed1,
 				randomSeed2,
 			);
 			validators.push(...standbyDelegates);
 		} else if (this._moduleConfig.numberStandbyDelegates === 1) {
-			const standbyDelegates = selectStandbyDelegates(snapshot.delegateWeightSnapshot, randomSeed1);
+			standbyDelegates = selectStandbyDelegates(snapshot.delegateWeightSnapshot, randomSeed1);
 			validators.push(...standbyDelegates);
 		}
 		const shuffledValidators = shuffleDelegateList(randomSeed1, validators);
-		await this._validatorsAPI.setGeneratorList(apiContext, shuffledValidators);
+		const bftValidators = [];
+		for (const validatorAddress of shuffledValidators) {
+			const validatorAccount = await this._validatorsAPI.getValidatorAccount(
+				apiContext,
+				validatorAddress,
+			);
+			const bftWeight =
+				standbyDelegates.findIndex(addr => addr.equals(validatorAddress)) > -1
+					? BigInt(0)
+					: BigInt(1);
+			bftValidators.push({
+				address: validatorAddress,
+				generatorKey: validatorAccount.generatorKey,
+				blsKey: validatorAccount.blsKey,
+				bftWeight,
```

Comment:
here if validator is not active, setting the bftWeight to be zero. Also, maintaining shuffled order